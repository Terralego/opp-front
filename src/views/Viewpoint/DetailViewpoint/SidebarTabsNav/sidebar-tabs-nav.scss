@import '../../../../variables';

.sidebar {

  // Override Blueprintjs
  @extend .override-tabs-bp3;

  .bp3-tabs {
    background-color: #0E5A8A
  }
}
