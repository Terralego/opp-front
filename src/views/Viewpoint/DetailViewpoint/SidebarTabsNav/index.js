import React, { useState } from 'react';
import { Tab, Tabs } from '@blueprintjs/core';
import { useTranslation } from 'react-i18next';

import DescriptionTab from '../SidebarTabsContent/DescriptionTab';
import InformationTab from '../SidebarTabsContent/InformationTab';

import './sidebar-tabs-nav.scss';
import '../SidebarTabsContent/sidebar-tabs-content.scss';

const TABS = 'Tabs__Sidebar';
const ID_DESCRIPTION_TAB = 'description_tab';
const ID_INFORMATION_TAB = 'information_tab';

export const SidebarTabsNav = () => {
  const [navTabId, setNavTabId] = useState(ID_INFORMATION_TAB);
  const { t } = useTranslation();

  return (
    <Tabs id={TABS} onChange={setNavTabId} selectedTabId={navTabId}>
      <Tab
        id={ID_INFORMATION_TAB}
        title={t('viewPoint.detail.information.title')}
        panel={<InformationTab />}
      />
      <Tab
        id={ID_DESCRIPTION_TAB}
        title={t('viewPoint.detail.description.title')}
        panel={<DescriptionTab />}
      />
    </Tabs>
  );
};

export default SidebarTabsNav;
