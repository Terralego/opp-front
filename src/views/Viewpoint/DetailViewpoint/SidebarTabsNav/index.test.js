import React from 'react';
import renderer from 'react-test-renderer';

import SidebarTabsNav from '.';


jest.mock('@blueprintjs/core', () => ({
  Tabs: ({ children }) => children,
  Tab: () => <p>Tab</p>,
}));

jest.mock('../../../../components/Visualizer/Visualizer', () => ({
  addCustomIcon: () => null,
}));

it('should render', () => {
  const tree = renderer.create((
    <SidebarTabsNav />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});
