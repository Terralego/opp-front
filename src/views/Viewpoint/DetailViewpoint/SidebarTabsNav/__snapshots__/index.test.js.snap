// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
Array [
  <p>
    Tab
  </p>,
  <p>
    Tab
  </p>,
]
`;
